manager/src/main/java/org/openremote/manager/dashboard/DashboardResourceImpl.java

```diff
@@ -30,31 +31,68 @@ public DashboardResourceImpl(TimerService timerService, ManagerIdentityService i
 
     @Override
     public Dashboard[] getAllRealmDashboards(RequestParams requestParams, String realm) {
-
+        boolean publicOnly;
+        if(realm == null) {
+            throw new WebApplicationException(BAD_REQUEST);
+        }
         try {
+            // if not authenticated, or having no INSIGHTS access, only fetch public dashboards
+            if(isAuthenticated()) {
+                publicOnly = (!hasResourceRole(ClientRole.READ_INSIGHTS.getValue(), Constants.KEYCLOAK_CLIENT_ID));
+            } else {
+                publicOnly = true;
```

Comment:
Just set `publicOnly = true` when creating the variable rather than use an else statement